src/core/server/server.ts

```diff
@@ -405,4 +439,92 @@ export class Server {
       this.configService.setSchema(descriptor.path, descriptor.schema);
     }
   }
+
+  private registerKibanaStartedEvent(analyticsSetup: AnalyticsServiceSetup) {
+    analyticsSetup.registerEventType<{ uptime_per_step: UptimeSteps }>({
+      eventType: KIBANA_STARTED_EVENT,
+      schema: {
+        uptime_per_step: {
+          properties: {
+            constructor: {
+              properties: {
+                start: {
+                  type: 'float',
+                  _meta: {
+                    description:
+                      'Number of seconds the Node.js process has been running until the constructor was called',
+                  },
+                },
+                end: {
```

Comment:
Just speaking my mind here: Are we planning on calculating the total _constructor, setup, start_ times at a later stage? Would it make sense to have this calculated here and having something like `elapsedTime` instead?

Reply:
I thought about that, but then I asked myself: would it make sense to have `duration` from `constructor` to `setup` or `start`?

I thought we can do that math as needed in the analysis end 😇